assets/blocks/course-outline/apply-parent-style.js

```diff
@@ -0,0 +1,126 @@
+import { select, dispatch } from '@wordpress/data';
+import TokenList from '@wordpress/token-list';
+import { find } from 'lodash';
+
+/**
+ * Checks if a block has a registered style which matches with the supplied class.
+ *
+ * @param {Array}  blockStyles The block's registered styles.
+ * @param {string} className   The class to check.
+ *
+ * @return {boolean} True if there is a match.
+ */
+const blockHasStyle = function ( blockStyles, className ) {
+	return (
+		blockStyles &&
+		blockStyles.some( ( style ) => 'is-style-' + style.name === className )
+	);
+};
+
+/**
+ * Returns the active style class from the given className.
+ *
+ * @param {Array}  styles    Block style variations.
+ * @param {string} className Class name
+ *
+ * @return {string} The active style class.
+ */
+export function getActiveStyleClass( styles, className ) {
+	let activeClass = null;
+
+	new TokenList( className ).forEach( ( potentialClass ) => {
+		if ( potentialClass.indexOf( 'is-style-' ) !== -1 ) {
+			const potentialStyleName = potentialClass.substring( 9 );
+			const activeStyle = find( styles, { name: potentialStyleName } );
+
+			if ( activeStyle ) {
+				activeClass = potentialClass;
+			}
+		}
+	} );
+
+	if ( ! activeClass ) {
+		const defaultStyle = find( styles, 'isDefault' );
+
+		activeClass = defaultStyle ? 'is-style-' + defaultStyle.name : null;
+	}
+
+	return activeClass;
+}
+
+/**
+ * Applies the style class of a parent block when the style is updated. The value of the style needs to be tracked by
+ * callers.
+ *
+ * @param {string} parentBlockName The name of the parent block.
+ * @param {string} childBlockName  The name of the child block.
+ * @param {string} childBlockId    The clientId of the block to apply the style to.
+ * @param {string} oldParentClass  The previous style class of the parent block.
+ * @param {string} onUpdate        A callback to be called when the the style of the parent block is updated.
+ */
+export const applyParentStyle = function (
```

Comment:
The thing is that we can't use `useSelect` and `useDispatch` inside an effect due to the limitations of hooks. There are probably ways around it but it would make the signature of the function more complicated so I think we are better keeping it like this.